routes/playlists.routes.js

```diff
@@ -0,0 +1,187 @@
+const express = require("express");
+const mongoose = require('mongoose');
+const {User} = require("../models/user.model")
+const {Playlist} = require("../models/playlist.model")
+
+const router = express.Router();
+
+router.route("/")
+.get(async(req,res) => {
+  try{
+    const userId = res.locals.userId;
+    console.log(userId)
+    const user = await User.findById(userId);
+    console.log(user);
+    const {playlists} = await User.findById(userId).populate("playlists");    
+    if(playlists){
+      res.status(200).json({success : true, data : {playlists}})
+    } else {
+      res.status(404).json({success:true, data : {}, message : "No user found with this ID"})  
+    }      
+  } catch (err) {
+    res.status(500).json({ success: false, message: "unable to find user playlists", errorMessage: err.message})
+  }  
+})
+
+router.route("/")
+.post(async(req,res) => {
+  try{
+    const userId  = res.locals.userId
+    const user = await User.findById(userId);
+    if(user){
+      const playlist = {        
+        name: req.body.name,
+        type : req.body.type,                
+      }
+      const newPlaylist = new Playlist(playlist);
+      const savedPlaylist = await newPlaylist.save();    
+      user.playlists.push(savedPlaylist._id);
+      const savedUser = await user.save();
+      res.status(201).json({success : true, data : { playlist : savedPlaylist }})
+    } else {
+      res.status(404).json({success:true, data : {}, message : "No user found with this ID"})
+    }    
+  } catch (err) {
+    res.status(500).json({ success: false, message: "unable to add playlist to user", errorMessage: err.message})
+  }  
+})
+
+router.route("/:playlistId")
+.get(async(req,res) => {
+  try{
+    const {playlistId} = req.params;
+    const playlist = await Playlist.findById(playlistId);
+    if(playlist){
+      res.status(200).json({success : true, data : { playlist : playlist }})
+    } else{
+      res.status(404).json({success:true, data : {}, message : "No playlist found with this ID"})
+    }
+  } catch (err) {
+      res.status(500).json({ success: false, message: "unable to find playlist", errorMessage: err.message})
+  }  
+})
+
+
+router.route("/:playlistId")
```

Comment:
Instead of creating a separate route chain it below the `.get` request as `route` is similar for the `.get` and `.post` req

Reply:
yeah no need of again doing router.route